hyperspy/io_plugins/edax.py

```diff
@@ -0,0 +1,943 @@
+
+    Returns
+    -------
+    list
+        list with dictionary of signal information to be passed back to
+        hyperspy.io.load_with_reader
+    """
+    with open(filename, 'rb') as f:
+        spd_header = np.fromfile(f,
+                                 dtype=get_spd_dtype_list(endianess),
+                                 count=1)
+
+        original_metadata = {'spd_header': sarray2dict(spd_header)}
+
+        # dimensions of map data:
+        nx = original_metadata['spd_header']['nPoints']
+        ny = original_metadata['spd_header']['nLines']
+        nz = original_metadata['spd_header']['nChannels']
+        data_offset = original_metadata['spd_header']['dataOffset']
+        data_type = {'1': 'u1',
+                     '2': 'u2',
+                     '4': 'u4'}[str(original_metadata['spd_header'][
+                         'countBytes'])]
+
+        mode = kwargs.pop('mode', 'c')
+
+        # Read data from file into a numpy memmap object
+        data = np.memmap(f, mode=mode, offset=data_offset, dtype=data_type,
+                         **kwargs).squeeze().reshape((nz, nx, ny), order='F').T
+
+    # Convert char arrays to strings:
+    original_metadata['spd_header']['tag'] = \
+        spd_header['tag'][0].view('S16')[0]
+    # fName is the name of the .bmp (and .ipr) file of the map
+    original_metadata['spd_header']['fName'] = \
+        spd_header['fName'][0].view('S120')[0]
+
+    # Get name of .spc file from the .spd map (if not explicitly given):
+    if not spc_fname:
+        spc_path = os.path.dirname(filename)
+        spc_basename = os.path.splitext(os.path.basename(filename))[
+            0] + '.spc'
+        spc_fname = os.path.join(spc_path, spc_basename)
+
+    read_spc = os.path.isfile(spc_fname)
+    if not read_spc:
+        _logger.warning('Could not find .spc file named {}.\n'
+                        'No spectral metadata will be loaded.'
+                        '\n'.format(spc_fname))
+
+    # Get name of .ipr file from bitmap image (if not explicitly given):
+    if not ipr_fname:
+        ipr_basename = os.path.splitext(
+            os.path.basename(
+                original_metadata['spd_header'][
+                    'fName']))[0].decode() + '.ipr'
+        ipr_path = os.path.dirname(filename)
+        ipr_fname = os.path.join(ipr_path, ipr_basename)
+
+    read_ipr = os.path.isfile(ipr_fname)
+    if not read_ipr:
+        _logger.warning('Could not find .ipr file named {}.\n'
+                        'No spatial calibration will be loaded.'
+                        '\n'.format(ipr_fname))
+
+    # Read the .ipr header (if possible)
+    if read_ipr:
+        with open(ipr_fname, 'rb') as f:
+            ipr_header = np.fromfile(f,
+                                     dtype=get_ipr_dtype_list(endianess),
+                                     count=1)
+            original_metadata['ipr_header'] = sarray2dict(ipr_header)
+
+    # Read the .spc header (if possible)
+    if read_spc:
```

Comment:
Yeah... I don't know why I did that. Hopefully the updated code makes more sense. Let me know if anything else needs changing.